blocks/basic/include/gnuradio-4.0/basic/DataSink.hpp

```diff
@@ -325,7 +325,8 @@ class DataSink : public Block<DataSink<T>> {
                 fnc(readData, std::span<const Tag>(relevantTags));
                 std::ignore = tag_reader.consume(relevantTags.size());
             } else {
-                std::ignore = tag_reader.consume(tag_reader.available());
+                const auto tags = tag_reader.get();
+                std::ignore     = tag_reader.consume(tag_reader.available());
```

Comment:
should probably be 

```cpp
if (tag_reader.consume(tags.size())) { /* handle error*/};
```

since `available()` may change between successive calls.